ext/page_log/palm/palm_kv.c

```
@@ -477,7 +477,7 @@ palm_kv_get_page_ids(
 
     *size = count;
 
-    return (ret);
+    return (0);
```

I wouldn't change this - if someone forgets to process their error, it's good not to silent it.

---

Currently returning `ret` here will always cause the function to return `MDB_NOTFOUND`. This error code indicates the end of the PALM walk, so it’s expected and not an error.
Since there’s already a condition on line 468 to return `ret` if it’s not `0` or `MDB_NOTFOUND`, and we return early on any real errors, I think it would be okay to return `0` here instead.
Another option would be to explicitly reset `ret` to `0` before returning to make it clear this is a success case, though that might be a bit overkill.

---

I agree that it's better to not change this line.

We would also like to ensure that we don't accidentally mask any error. You should add a check whether the return code is `MDB_NOTFOUND`, in which case you would reset it to 0.

---

Sounds good, I've reset ret to 0 for the success case and changed the return to 0 in [7f684f1](https://github.com/wiredtiger/wiredtiger/pull/12306/commits/7f684f1d7cbbbd3572c557f7e0e2f20be8a1a016)